include/small/DarknetParser.hpp

```diff
@@ -0,0 +1,511 @@
+        if(line.empty() || line.at(0) == '#' ) {
+            continue;
+        }
+        // we hit the next block
+        if(line.at(0) == '[') {
+            cfg_file->seekg(last_pos); // make sure to move file ptr back
+            break;
+        }
+
+        // parse the line
+        std::string key = line.substr(0, line.find("="));
+        std::string value = line.substr(line.find("=") + 1);
+        if(key == "size") {
+            block_info.size = std::stoi(value);
+        } else if(key == "stride") {
+            block_info.stride = std::stoi(value);
+        } else if(key == "pad") {
+            // todo: check this
+            block_info.pad = value == "1" ?  small::PADDING_F : small::PADDING_V;
+        } else if(key == "filters" || key == "channels") {
+            block_info.c = std::stoi(value);
+        } else if(key == "activation") {
+            block_info.activation = get_activation(value);
+        } else if(key == "batch_normalize") {
+            block_info.bn = std::stoi(value);
+        } else if(key == "layers" || key == "from") {
+            block_info.layers = extract_int_array<int>(value);
+        } else if(key == "mask") {
+            block_info.mask = extract_int_array<uint32_t>(value);
+        } else if(key == "anchors") {
+            block_info.anchors = get_anchors(value);
+        } else if(key == "jitter") {
+            block_info.jitter = std::stof(value);
+        } else if(key == "ignore_thresh") {
+            block_info.ignore_thresh = std::stof(value);
+        } else if(key == "truth_thresh") {
+            block_info.truth_thresh = std::stof(value);
+        } else if(key == "random") {
+            block_info.random = std::stoi(value);
+        } else if(key == "classes") {
+            block_info.classes = std::stoi(value);
+        } else if(key == "num") {
+            block_info.num = std::stoi(value);
+        } else if(key == "width") {
+            block_info.w = std::stoi(value);
+        } else if(key == "height") {
+            block_info.h = std::stoi(value);
+        } else {
+            // #if PARSER_DEBUG == 1
+            // std::cout << "Unknown key: " << key << std::endl;
+            // #endif
+            // exit(1);
+        }
+
+        last_pos = cfg_file->tellg();
+    }
+
+    return block_info;
+}
+
+template <typename BufferT>
+small::Layer<BufferT>* parse_conv(small::shape_type &input, block_info_t block_info, float *weights) {
+    
+    size_t filt_size = input[small::CHANNEL]*block_info.size*block_info.size*block_info.c;
+
+    #if PARSER_DEBUG == 1
+    std::cout << "Parsing conv layer with filt size: " << filt_size << std::endl;
+    if(block_info.bn) {
+        std::cout << "Batch norm weights: " << block_info.c*4 << std::endl;
+    }
+    else {
+        std::cout << "Conv bias: " << block_info.c << std::endl;
+    }
+    #endif
+
+    BufferT biases(block_info.c);
+
+    // has batch norm weights
+    if(block_info.bn) {
```

Comment:
Need to discuss possibility of two bias vectors, and also BufferT's go out of scope and get deleted.

Reply:
Oh whoops, I will apply a fix for the scoping issue.

2 bias vectors? I haven't seen that before. Do you have a site that showcases that possibility?

Reply:
I haven't handled this case in my newest changes yet. This needs to be discussed further.